### PR TITLE
Add support for Amazon Linux aarch64 Dockerfile for main nightly branch

### PR DESCRIPTION
**Pull request by shahmishal**

### CODE REVIEW

**Comment by shahmishal** [comment]
@swift-ci test

File: nightly-main/amazonlinux/2/buildx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+FROM base AS base-arm64
+ARG OS_ARCH_SUFFIX=-aarch64
+ARG ADDITIONAL_TAR_OPTIONS=
+
+FROM base-$TARGETARCH AS final
+
+ARG OS_VER=$SWIFT_PLATFORM$OS_MAJOR_VER$OS_ARCH_SUFFIX
+ARG PLATFORM_WEBROOT="$SWIFT_WEBROOT/$SWIFT_PLATFORM$OS_MAJOR_VER$OS_ARCH_SUFFIX"
+
+RUN echo "${SWIFT_WEBROOT}/latest-build.yml"
```
Comment (fabianfett) [inline nightly-main/amazonlinux/2/buildx/Dockerfile]:
```suggestion
RUN echo "${PLATFORM_WEBROOT}/latest-build.yml"
```

File: nightly-main/amazonlinux/2/buildx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    && export DOWNLOAD_DIR=$(echo $download | sed "s/-${OS_VER}.tar.gz//g") \
+    && echo $DOWNLOAD_DIR > .swift_tag \
+    # - Download the GPG keys, Swift toolchain, and toolchain signature, and verify.
+    && export GNUPGHOME="$(mktemp -d)" \
+    && curl -fsSL ${PLATFORM_WEBROOT}/${DOWNLOAD_DIR}/${download} -o latest_toolchain.tar.gz \
+    ${PLATFORM_WEBROOT}/${DOWNLOAD_DIR}/${download_signature} -o latest_toolchain.tar.gz.sig \
+    && curl -fSsL https://swift.org/keys/all-keys.asc | gpg --import -  \
+    && gpg --batch --verify latest_toolchain.tar.gz.sig latest_toolchain.tar.gz \
+    # - Unpack the toolchain, set libs permissions, and clean up.
+    && tar -xzf latest_toolchain.tar.gz --directory / --strip-components=1 \
```
Comment (fabianfett) [inline nightly-main/amazonlinux/2/buildx/Dockerfile]:
```suggestion
    && tar -xzf latest_toolchain.tar.gz --directory / ${ADDITIONAL_TAR_OPTIONS} \
```

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
@swift-ci test

**Pull request merged**